backend.py

```diff
@@ -25,24 +62,42 @@ def __init__(self, direction, path, coordinates):
     def __repr__(self):
         return "<Robot {} {} {}>".format(self.direction, self.path, self.coordinates)
 
-    def walk(self, distance):
+    def walk(self, distance, state):
         """
         Move a robot to new coordinates based on its direction.
         """
 
-        self.move(self.direction, distance)
+        self.move(self.direction, distance, state)
 
-    def move(self, direction, distance):
+    def move(self, direction, distance, state):
         """
         Move a robot to new coordinates according to direction of the move.
         """
-
-        (x, y) = self.coordinates
-        (new_x, new_y) = direction.coor_delta
-        x = x + (new_x * distance)
-        y = y + (new_y * distance)
-
-        self.coordinates = (x, y)
+        for __ in range(distance):
+            old_tiles = state.board[self.coordinates]
+            # On the current tile; check wall in the direction of next move.
+            for tile in old_tiles:
+                tile_move = tile.can_move_from(direction)
+                if tile_move is False:
+                    break
+            if tile_move:
+                # There is no wall, so get new coordinates
+                (x, y) = self.coordinates
+                (new_x, new_y) = direction.coor_delta
+                x = x + new_x
+                y = y + new_y
+                new_tiles = state.board[(x, y)]
+                # Check wall on the next tile in the direction of the robot movement.
+                for tile in new_tiles:
+                    tile_move = tile.can_move_to(direction)
+                    if tile_move is False:
+                        break
+                if tile_move:
+                    self.coordinates = (x, y)
+                else:
+                    break
+            else:
+                break
```

Comment:
Jsem ztracená u def_properties, když tam pod tím máš hned napsané že dostaneš type a vracíš types.  I když vidím že v budoucnu bude vracet více properties tak mě to mate teda a nevím :-D

Reply:
Chápu, že to zatím vypadá zmateně, ale chci to v dohledné době dodělat/předělat.